fix(redraw): avoid unnecessary redraws and glitches with floats+messages

bfredl opened the pull request:
fixes https://github.com/neovim/neovim/issues/20106 and maybe others. will go through all recent cmdheight issues and write tests.

bfredl committed:
fix(redraw): make redrawdebug=nodelta handle all the cases

Before only win_line lines were considered. this applies nodelta
to all screen elements. Causes some failures, which might indeed
indicate excessive redraws.

bfredl committed:
fix(redraw): avoid unnecessary redraws and glitches with floats+messages

fixes #20106
fixes #20229

Shougo commented:
I have tested it.

It fixes #20106 #20229 #20243

Shougo commented:
It is better.  But it is not perfect.

It clears current command line like this.

https://github.com/neovim/neovim/pull/20233#issuecomment-1250084410

zeertzjq commented:
> It clears current command line like this.

Does <ttps://github.com/vim/vim/commit/a653e53b1f4ba08de0dbcea06288cf0cc1c6e752> fix this?

Shougo commented:
```diff
diff --git a/src/nvim/ex_docmd.c b/src/nvim/ex_docmd.c
index 52841b602..7d126882d 100644
--- a/src/nvim/ex_docmd.c
+++ b/src/nvim/ex_docmd.c
@@ -6077,6 +6077,11 @@ static void ex_redraw(exarg_T *eap)
   // No need to wait after an intentional redraw.
   need_wait_return = false;
 
+  // When invoked from a callback or autocmd the command line may be active.
+  if (State & MODE_CMDLINE) {
+    redrawcmdline();
+  }
+
   ui_flush();
 }
 ```

Unfortunately, no...

bfredl committed:
fix(tests): indicate in test logs when nvim exit times out

When it happens it wastes 2 seconds which is NOT included in the
normal busted timing info. It is hard to correct this, but we
can at least print a warning when this happens.

justinmk commented on runtime/lua/man.lua:
```diff
@@ -23,13 +23,15 @@ local function man_system(cmd, silent)
 
-  local handle = vim.loop.spawn(cmd[1], {
+  local handle
+  handle = vim.loop.spawn(cmd[1], {
```
off-topic: man_system seems like something that would go in the stdlib.
